File: 5.des/freshIngredients.py
def solve_inventory(input_text):
    lines = input_text.strip().split('\n')
    blank_line_index = lines.index('')
    
    range_lines = lines[:blank_line_index]
    ranges = []
    for line in range_lines:
        min_val, max_val = line.split('-')
        ranges.append({'min': int(min_val), 'max': int(max_val)})
    
    id_lines = lines[blank_line_index + 1:]
    ids = [int(line) for line in id_lines]
    
    fresh_count = 0
    for id_num in ids:
        is_fresh = any(
            id_num >= range_dict['min'] and id_num <= range_dict['max']
            for range_dict in ranges
        )
        if is_fresh:
            fresh_count += 1
    
    return fresh_count

with open("input.txt", "r") as file:
    input_text = file.read()

print(solve_inventory(input_text))

File: 5.des/freshIngredients.ts
import * as fs from "fs";
//taka inn string sklia number
function solveInventory(input: string): number {
  // split input, array
  const lines = input.trim().split('\n');
  
  // Staðsetja blank line
  const blankLine = lines.findIndex(line => line.trim() === '');
  console.log(blankLine)
  
  // Sækja allar línur að blankLine
  const rangeLines = lines.slice(0, blankLine);
  //fylki sem breyir range i min og max
  const ranges: Array<{min: number, max: number}> = rangeLines.map(line => {
    //breyta i strings og svo number og gef min og max
    const [min, max] = line.split('-').map(Number);
    return { min, max };
  });
  
  //svo takka öll ingredients id (eftir blankline) og breyta svo i number
  const idLines = lines.slice(blankLine + 1);
  const ids = idLines.map(Number);
  
  // Check each ID to see if it's fresh
  let freshCount = 0;
  
  for (const id of ids) {
    // id er fresh ef það sé i eitthvað range (some() skilar true ef 1 er i range)
    const isFresh = ranges.some(range => 
      id >= range.min && id <= range.max
    );
    
    if (isFresh) {
      freshCount++;
    }
  }
  
  return freshCount;
}

//lesa txt
const input = fs.readFileSync("input.txt", "utf-8")

console.log(solveInventory(input))